go-controller/pkg/util/net_unit_test.go

```diff
@@ -91,33 +94,33 @@ func TestGetPortAddresses(t *testing.T) {
 		{
 			desc:                    "test code path where IP address parsing fails",
 			input:                   "TEST_PORT",
-			errExpected:             true,
+			errExpected:             false,
 			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("22:e9:ac:f4:00:04 10.244.0.\n[dynamic]")), nil, nil}},
 			onRetArgsKexecIface:     &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
 		},
 		{
 			desc:                    "test code path where addresses list count is less than 2",
 			input:                   "TEST_PORT",
-			errExpected:             true,
+			errExpected:             false,
 			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{nil, nil, nil}},
 			onRetArgsKexecIface:     &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
 		},
-		{
-			desc:                    "test code path when RunOVNNbctl returns only static address",
-			input:                   "TEST_PORT",
-			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("[]\n[\"06:c6:d4:fb:fb:ba 10.244.2.2\"]")), nil, nil}},
-			onRetArgsKexecIface:     &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
-		},
-		{
-			desc:                    "test the code path when RunOVNNbctl returns `[]\\n[dynamic]`",
-			input:                   "TEST_PORT",
-			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("[]\n[dynamic]")), nil, nil}},
-			onRetArgsKexecIface:     &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
-		},
+		// {
```

Comment:
FYI: The test `TestGetPortAddresses` might have to be re-written to look something like https://github.com/vjayaramrh/ovn-kubernetes/blob/vish_abhat/go-controller/pkg/util/net_unit_test.go#L64. (I pulled your PR locally to see what the modifications would look like for the unit test for this function. I also had to create additional mocks from the interface for the `Client` defined in ebay/go-ovn.  The link at `https://github.com/ovn-org/ovn-kubernetes/compare/master...vjayaramrh:vish_abhat` will show changes that I made to come up with the modified unit test.

Reply:
When writing the test case, I could really see the benefit of using the go-ovn bindings. Is there a go-ovs binding project as well that could replace some of the OVS cli functions that we are invoking currently in the ovs.go file?

Reply:
Came across this link https://github.com/digitalocean/go-openvswitch , however, not sure if this gives us the go-ovs bindings that our project might need.

Reply:
We had looked at that library too. It's less actively maintained than the ebay one.

Reply:
So I presume that the ebay/go-ovn supports OVS go-bindings as well, right.? I was of the impression that it supported only the OVN go-bindings.

Reply:
Not via the Client interface. But libovsdb can do that work for us.